frontend/src/utilities/useFetchState.stories.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { useCallback } from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import { within, userEvent } from '@storybook/testing-library';
+import { expect } from '@storybook/jest';
+import { Icon, Spinner } from '@patternfly/react-core';
+import { CloseIcon } from '@patternfly/react-icons';
+import useFetchState from './useFetchState';
+
+type TestComponentProps<T> = {
+  defaultState: T;
+  response: T;
+  refreshRate?: number;
+  responseFailed: boolean;
+  responseError?: Error;
+  responseDelay: number;
+  fetchCallbackPromise: () => Promise<T>;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-constraint
+const TestComponent = <T extends unknown>({
+  defaultState,
+  response,
+  refreshRate,
+  responseFailed = false,
+  responseError,
+  responseDelay = 0,
+  fetchCallbackPromise,
+}: TestComponentProps<T>) => {
+  // pick func to use and apply delay
+  const callback = useCallback(async () => {
+    await new Promise((resolve) => setTimeout(resolve, responseDelay));
+
+    if (fetchCallbackPromise) {
+      return fetchCallbackPromise();
+    }
+
+    if (responseFailed) {
+      throw responseError;
+    }
+    return Promise.resolve(response);
+  }, [fetchCallbackPromise, responseDelay, responseError, response, responseFailed]);
+  // hook to test
+  const [result, loaded, loadError, refresh] = useFetchState(callback, defaultState, refreshRate);
+
+  return (
```

Comment:
So I kinda envisioned a generic component 🤔 

Something that maybe rendered like:

```
Parameters              Response (Array) [unstable]
------------            --------------------
Function [stable]       Object ({ ... }) [stable] //*
Array ([]) [unstable]   boolean (false)
number (#)              undefined
                        Function [stable]


// * maybe some collapse with a click to expand details?
```

Where the component rendering is generic for all hooks. Maybe we even wrap our test hook in another hook that tracks ref stability of each value. You could do that with a simple setState value in the component -- causing another render and talking about "is the old reference the same as the new one"...

Might be worth just building something like this to help with testing hooks and handling referential stability.  Referential stability is a huge thing in hooks and I think it would help to showcase that information in tests for the given hook. Especially for the response values -- as that can cause issues under the hood (frequent component refreshes / side-effects) for all the various users of the hook. Some might care far less than others and I think the test can help get us aligned on making things work well.

Let me know if you want to talk about this.

These test work, so we could go forward with this -- but I think we need to consider a proper infra for future hooks.

I dunno, maybe this is overkill ... would be nice to be able to test hooks without having to write a custom renderer and deal with items (like stability) every time.